sideband: init at 1.3.0

GaetanLepage opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

Add [sideband](https://github.com/markqvist/Sideband), an LXMF client allowing you to communicate with people or LXMF-compatible systems over Reticulum networks.

- **python312Packages.ffpyplayer: init at 4.5.2**
- **python312Packages.pycodec2: init at 4.0.0**
- **python312Packages.pyobjus: init at 1.2.3**
- **sideband: init at 1.3.0**

---

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
sideband: init at 1.3.0

GaetanLepage committed:
python312Packages.ffpyplayer: init at 4.5.2

GaetanLepage committed:
python312Packages.pycodec2: init at 4.0.0

GaetanLepage committed:
python312Packages.pyobjus: init at 1.2.3

adingbatponder commented:
It would be great to make this work.... thanks so much for the efforts.

Erethon commented:
Thank you for packaging sideband! A mini initial review:
1) `sideband -d` works on NixOS as is!
2) Running sideband with the UI enabled however currently fails. This is because `.kv` files that are needed for the app to work aren't included in the build. I'm not sure if this is because of some Nix sandboxing, but I've opened https://github.com/markqvist/Sideband/pull/72 upstream to look further into it. Applying this patch via a `patches` fixes this and sideband with the UI loads fine.
3) Kivy tries to find `xclip` or `xsel`, if you don't have any of these installed some errors get printed on the logs. Setting [xclip/xsel as a dependency with this patch](https://gist.github.com/Erethon/011b0c0529100831ebcf1c028d9e13e5) fixes this, but I'm not sure if this should be done on Sideband or on the Kivy package instead.

Ralith commented:
I unfortunately duplicated this effort in https://github.com/NixOS/nixpkgs/pull/421656. On the plus side, my version addresses the above issues.

GaetanLepage commented:
> I unfortunately duplicated this effort in #421656. On the plus side, my version addresses the above issues.

Great! Let's keep working in your PR then!
